eth/utils/bitfield.py

```diff
@@ -0,0 +1,41 @@
+from typing import (
+    List,
+)
+
+
+def has_voted(bitfield: bytes, index: int) -> bool:
+    return bool(bitfield[index // 8] & (128 >> (index % 8)))
+
+
+def set_voted(bitfield: bytes, index: int) -> bytes:
+    byte_index = index // 8
+    bit_index = index % 8
+    new_byte_value = bitfield[byte_index] | (128 >> bit_index)
+    return bitfield[:byte_index] + bytes([new_byte_value]) + bitfield[byte_index + 1:]
+
+
+def get_bitfield_length(bit_count: int) -> int:
+    """Return the length of the bitfield for a given number of attesters in bytes."""
+    return (bit_count + 7) // 8
+
+
+def get_empty_bitfield(bit_count: int) -> bytes:
+    return b"\x00" * get_bitfield_length(bit_count)
+
+
+def get_vote_count(bitfield: bytes) -> int:
+    votes = 0
+    for index in range(len(bitfield) * 8):
+        if has_voted(bitfield, index):
+            votes += 1
```

Comment:
Non mutative approach:

```python
# make has_voted curriable
@curry
def has_voted(bitfield, index):
    ...

return len(tuple(filter(has_voted(bitfield), (idx for idx in range(len(bitfield) * 8)))))
```

psuedo-code but it should convey the idea.  The `filter` could be replaced with a comprehension using `[v for v in thing if has_voted(...)]` which might increase readability.  Also line-breaks and indentation can improve readability a lot here.